chain/jsonrpc-primitives/src/types/query.rs

```diff
@@ -0,0 +1,151 @@
+                        })?,
+                    },
+                },
+                "code" => near_primitives::views::QueryRequest::ViewCode { account_id },
+                "contract" => near_primitives::views::QueryRequest::ViewState {
+                    account_id,
+                    prefix: data.into(),
+                },
+                "call" => match maybe_extra_arg {
+                    Some(method_name) => near_primitives::views::QueryRequest::CallFunction {
+                        account_id,
+                        method_name: method_name.to_string(),
+                        args: data.into(),
+                    },
+                    None => {
+                        return Err(crate::errors::RpcParseError(
+                            "Method name is missing".to_string(),
+                        ))
+                    }
+                },
+                _ => {
+                    return Err(crate::errors::RpcParseError(format!(
+                        "Unknown path {}",
+                        query_command
+                    )))
+                }
+            };
+            RpcQueryRequest {
+                request,
+                block_reference: near_primitives::types::BlockReference::latest(),
+            }
+        } else {
+            crate::utils::parse_params::<RpcQueryRequest>(value)?
+        };
+        Ok(query_request)
+    }
+}
+
+impl From<near_client_primitives::types::QueryError> for RpcQueryError {
+    fn from(error: near_client_primitives::types::QueryError) -> Self {
+        match error {
+            near_client_primitives::types::QueryError::IOError(s) => Self::InternalError(s),
+            near_client_primitives::types::QueryError::NotSyncedYet => Self::NotSyncedYet,
+            near_client_primitives::types::QueryError::DoesNotTrackShard => Self::DoesNotTrackShard,
+            near_client_primitives::types::QueryError::InvalidAccount(account_id) => {
+                Self::InvalidAccount(account_id)
+            }
+            near_client_primitives::types::QueryError::AccountDoesNotExist(account_id) => {
+                Self::AccountDoesNotExist(account_id)
+            }
+            near_client_primitives::types::QueryError::ContractCodeDoesNotExist(account_id) => {
+                Self::ContractCodeDoesNotExist(account_id)
+            }
+            near_client_primitives::types::QueryError::AccessKeyDoesNotExist(public_key) => {
+                Self::AccessKeyDoesNotExist(public_key)
+            }
+            near_client_primitives::types::QueryError::VMError(s) => Self::VMError(s),
+            near_client_primitives::types::QueryError::Unreachable(s) => {
+                near_metrics::inc_counter_vec(
+                    &crate::metrics::RPC_UNREACHABLE_ERROR_COUNT,
+                    &["RpcQueryError", &s],
+                );
+                Self::Unreachable(s)
```

Comment:
I recall we had some discussion about adding logging for such unreachable states, so not only we can see the counter, but actually address those based on which case we missed. Do we log it anywhere?